### DISCUSSION TITLE
Add to Alpine.store which may or may not exist yet

### DISCUSSION THREAD

**Discussion by anjanesh** [alpinejs/alpine#3564] [1. Help]
I have 2 JavaScript files - I won't know in which order they are coming.
So how do I add entries to `global_data` which may or may not exist ?
```
<script>
document.addEventListener('alpine:init', () =>
{
    Alpine.store('global_data', {
    foo1: 'bar1'    
    });
});

document.addEventListener('alpine:init', () =>
{
    Alpine.store('global_data', {
    foo2: 'bar2'    
    });
});
</script>

<div x-text="$store.global_data.foo1"></div>
<div x-text="$store.global_data.foo2"></div>
```

**Comment by ekwoka** [comment]
You just don't do this.

If you can't possibly bundle this stuff together, then the best way would be to do something like


```js
window.global_data = {}

document.addEventListener('alpine:init', () => {
   window.global_data = Alpine.reactive(window.global_data);
   Alpine.store('global_data', window.global_data)
}
```

Then you can just add values on the global data from anywhere with no worries.

Though, this does also get rid of the need to use `$store` at all, but I guess that's fine too.
